### PR TITLE
Reorg manual tracing JS docs based on SDK

### PR DESCRIPTION
**Pull request by cartermp**
This is built on https://github.com/open-telemetry/opentelemetry.io/pull/1435 out of convenience, but I'm unopinionated on which order to pull in things.

Fixes https://github.com/open-telemetry/opentelemetry.io/issues/1436

I had to kind of reorg the manual instrumentation docs a bit for JS here. I also decided to pull in some text from other docs to clarify some sections, and linkify some concepts as well now that we have language-agnostic concepts.

Happy to break stuff out if that helps with review.

Since y'all aren't code owners yet, tagging @open-telemetry/javascript-approvers to take a look. I have no doubt that there may be a few technical inaccuracies here that I'd like to catch before merging 🙂 

---

Preview: https://deploy-preview-1445--opentelemetry.netlify.app/docs/instrumentation/js/instrumentation/

### CODE REVIEW

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 1]
```diff
@@ -3,135 +3,185 @@ title: Instrumentation
...
+Next, ensure that `tracing.js` is required in your node invocation. For example:
+
+```
+node --require './tracing.js' <other-app-file.js>
+```
 
-As you have learned in the previous [Getting Started](../getting-started) guide you
-need a TracerProvider and an Exporter. Install the dependencies and add them to
-the head of your application code to get started:
+**It is required to `--require` your tracing initialization code to trace.**
```
Comment (vmarchaud) [inline content/en/docs/instrumentation/js/instrumentation.md:53]: I think i would reword this to take into account two things:
- This only concern CJS module (even though we don't support right now ESM but it will come)
- This is only required if you want auto-instrumentations because they monkey-patch 3rd party modules

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 2]
```diff
@@ -3,135 +3,185 @@ title: Instrumentation
...
+import { ZoneContextManager } from '@opentelemetry/context-zone';
 
-const provider = new BasicTracerProvider();
+const provider = new WebTracerProvider();
 
-// Configure span processor to send spans to the exporter
-provider.addSpanProcessor(new SimpleSpanProcessor(new ConsoleSpanExporter()));
-provider.register();
+provider.register({
+  // Changing default contextManager to use ZoneContextManager - supports asynchronous operations - optional
```
[the remark this answers is not in the document]
Comment (vmarchaud) [reply content/en/docs/instrumentation/js/instrumentation.md:77]: Yes, users will get the most value by using the ZoneContextManager however i think its important to explain that it won't support apps that compiles to >= ES6

[the remark this answers is not in the document]
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:77]: Gotcha. I'm just going to remove this part for now. Maybe it can be handled in a more in-depth way later.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 3]
```diff
@@ -3,135 +3,185 @@ title: Instrumentation
...
-## Initializing a Tracer
+It's recommended to have a separate `tracing.js` file that has all the SDK
+initialization code in it:
+
+```js
+import opentelemetry from "@opentelemetry/api";
+import nodeOTel from "@opentelemetry/sdk-node";
+
+const sdk = new nodeOTel.NodeSDK({
+  traceExporter: new opentelemetry.tracing.ConsoleSpanExporter(),
```
[the remark this answers is not in the document]
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:35]: For performance-sensitive scenarios, I would assume it's better to use sdk-trace-base?

[the remark this answers is not in the document]
Comment (vmarchaud) [reply content/en/docs/instrumentation/js/instrumentation.md:35]: Indeed but i think most users still want to use the node-sdk because of the auto instrumentation, i was just saying that we might want to document that it has a performance impact

[the remark this answers is not in the document]
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:35]: Are there figures somewhere around what the rough overhead might look like? I'm hesitant to say there's a performance impact without an idea of what kind of impact.

**Comment by dyladan** [comment]
I added comments on #1435 but reviewing two documents which both touch a lot of things is difficult to know which PR the comment "belongs" on. Some of those comments may be addressed here?

**Comment by cartermp** [comment]
Thanks @dyladan - and sorry for the confusion. I'll close #1435 and bring your feedback over to address

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 4]
```diff
@@ -3,135 +3,180 @@ title: Instrumentation
...
-  SimpleSpanProcessor,
-} = require('@opentelemetry/sdk-trace-base');
-const opentelemetry = require('@opentelemetry/api');
+import opentelemetry from "@opentelemetry/api";
+import { WebTracerProvider } from '@opentelemetry/sdk-trace-web';
 
-const provider = new BasicTracerProvider();
+const provider = new WebTracerProvider();
 
-// Configure span processor to send spans to the exporter
```
Comment (cartermp) [inline content/en/docs/instrumentation/js/instrumentation.md:49]: (needs to be brought back in likely)

> Batch span processor makes sense in some cases (typical server app) but not in others (serverless). It might make sense to add a comment about this because a serverless container may be killed before a batch is exported.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 5]
```diff
@@ -3,135 +3,180 @@ title: Instrumentation
...
 export const tracer = opentelemetry.trace.getTracer(
-  'example-basic-tracer-node'
+  'name-of-tracer-goes-here'
 );
 ```
 
-This registers a tracer provider with the OpenTelemetry API as the global tracer provider, and exports a tracer instance that you can use to create spans.
-
-If you do not register a global tracer provider, any instrumentation calls will be a no-op, so this is important to do!
+You'll need to bundle this file with your web application to be able to use
```
Comment (cartermp) [inline content/en/docs/instrumentation/js/instrumentation.md:86]: > This registers a global `TracerProvider` with the OpenTelemetry API, which may be used to create spans.
If no global `TracerProvider` is registered, instrumentations which use the API will use a no-op tracer and no spans will be created or exported.

I'll work out a way to incorporate this.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 6]
```diff
@@ -292,3 +334,83 @@ try {
...
+  for (let i = 0; i < 3; i += 1) {
+    doWork(parentSpan, i);
+  }
+
+  // Be sure to end the parent span!
+  parentSpan.end();
+}
+
+function doWork(parent, i) {
+  // To set a child span, we need to mark the parent span as the active span
```
Comment (cartermp) [inline content/en/docs/instrumentation/js/instrumentation.md:391]: From @dyladan:

> Not obvious what "set a child span" means for those not already very familiar with tracing.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 7]
```diff
@@ -188,27 +227,27 @@ const { SemanticAttributes } = require('@opentelemetry/semantic-conventions');
...
+
+  // Make sure to end the span if you're done with it!
   span.end();
 }
 ```
 
 ## Span events
 
-An event is a human-readable message attached to a span that represents "something happening" during its lifetime. You can think of it like a primitive log.
+A [Span Event](/docs/concepts/signals/traces/#span-events) is a human-readable
```
Comment (cartermp) [inline content/en/docs/instrumentation/js/instrumentation.md:247]: > IMO "something happening" is too general. I prefer to think of events as an activity that is tracked as a single point in time rather than a start time and and end time. For example, a page load may take time and be represented by a span, but the moment a page becomes interactive is a singular point in time and may be represented as a span event.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 8]
```diff
@@ -252,37 +295,36 @@ function someFunction(spanToLinkFrom) {
...
+    });
   }
-
-  span.end();
 }
+
+span.end();
 ```
 
+By default, the status for all spans is `Unset` rather than `Ok`. It is
```
Comment (cartermp) [inline content/en/docs/instrumentation/js/instrumentation.md:319]: > Might be a good idea to explain why this is not necessary?

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Flarna) [reply content/en/docs/instrumentation/js/instrumentation.md:35]: The overhead depend on the application. If it is using promises/await a lot and has mostly no CPU intensive tasks overhead is higher. Therefore it's impossible to provide any numbers.

> For performance-sensitive scenarios, I would assume it's better to use sdk-trace-base?

Not that instrumentations depend on a working context manager as they get current active span from context. Therefore if one decides to improve performance by using sdk-trace-base they have to be aware that most instrumentations wont work or the user has to take care to implement a context manager.

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 7 above)
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:247]: I will reword this a bit and then open a separate PR that gives a motivating wordy example in the span event section in language-agnostic conceptual docs.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 9]
```diff
@@ -252,37 +315,36 @@ function someFunction(spanToLinkFrom) {
...
+typically the job of another component in your telemetry pipeline to interpret
+the `Unset` status of a span, so it's best not to override this unless you're
```
Comment (legendecas) [inline content/en/docs/instrumentation/js/instrumentation.md:341]: This statement sounds to me like suggesting users to not set the status code OK. Am I understanding this correctly?

**Review by legendecas — COMMENTED** [review]
Nice work! Just a few comments.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 10]
```diff
@@ -3,135 +3,202 @@ title: Instrumentation
...
 
-In the following this guide will use the following sample app:
+To start [tracing](/docs/concepts/signals/traces/#tracing-in-opentelemetry),
+you'll need to have an initialized
+[`TracerProvider`](/docs/concepts/signals/traces/#tracer-provider) that will let
+you create a [`Tracer`](/docs/concepts/signals/traces/#tracer).
 
-```javascript
-'use strict';
+If a `TracerProvider` is not created, either explicitly or via SDK APIs, then
```
[the remark this answers is not in the document]
Comment (dyladan) [reply content/en/docs/instrumentation/js/instrumentation.md:16]: I agree

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 11]
```diff
@@ -3,135 +3,202 @@ title: Instrumentation
...
-  // simulate some random work.
-  for (let i = 0; i <= Math.floor(Math.random() * 40000000); i += 1) {}
-}
+To initialize tracing with the Node.js SDK, first ensure you have the SDK
+package and OpenTelemetry API installed:
+
+```shell
+npm install \
+  @opentelemetry/api \
+  @opentelemetry/sdk-node
```
Comment (dyladan) [inline content/en/docs/instrumentation/js/instrumentation.md:28]: sdk-node is an experimental package. I wouldn't recommend it to be the official website documentation without a disclaimer about that. I would also not remove the documentation which uses the stable packages

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 12]
```diff
@@ -3,135 +3,202 @@ title: Instrumentation
...
+Next, create a separate `tracing.js` file that has all the SDK initialization
+code in it:
 
-As you have learned in the previous [Getting Started](../getting-started) guide you
-need a TracerProvider and an Exporter. Install the dependencies and add them to
-the head of your application code to get started:
+```js
+import opentelemetry from "@opentelemetry/api";
+import nodeOTel from "@opentelemetry/sdk-node";
+import { ConsoleSpanExporter } from '@opentelemetry/sdk-trace-base';
```
Comment (dyladan) [inline content/en/docs/instrumentation/js/instrumentation.md:37]: importing from a package you did not ask the user to install

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 13]
```diff
@@ -3,135 +3,202 @@ title: Instrumentation
...
-```shell
-npm install @opentelemetry/api
-npm install @opentelemetry/sdk-trace-base
+const exporter = new ConsoleSpanExporter();
+
+const sdk = new nodeOTel.NodeSDK({
+  traceExporter: exporter,
+});
+
+sdk.start()
```
Comment (dyladan) [inline content/en/docs/instrumentation/js/instrumentation.md:45]: important to know that as of right now the `sdk.start()` method is async and it is possible that if a span is created in the next milliseconds it may not be initialized early enough to catch it.

**Review by dyladan — COMMENTED** [review]
Thanks for consolidating to a single PR

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 11 above)
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:28]: A quirk to this is that the getting-started docs use this SDK as of this PR: https://github.com/open-telemetry/opentelemetry.io/pull/697

Should that get reverted as well?

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 12 above)
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:37]: It's pulled in transitively - should that be called out explicitly? I'd have the same question for `@opentelemetry/sdk-trace-node` should that be used here instead.

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 9 above)
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:341]: Yep. It's at least my understanding that unless you're an instrumentation library author, you shouldn't set the status to `Ok`, but leave it as `Unset` for a processor or backend to interpret.

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 11 above)
Comment (dyladan) [reply content/en/docs/instrumentation/js/instrumentation.md:28]: It is the easiest way to get started so I'm not sure. It isn't likely to cause problems, but it is very likely to have changes which break its interface which users may not appreciate if it is presented as the preferred solution. It might be better to have it in a different documentation section explicitly marked as experimental.

Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:28]: I'll stick with `sdk-trace-node` for now and leave a note about how you can optionally use the `sdk-node` experimental package. When that package is no longer experimental we can update docs here.

**Comment by cartermp** [comment]
Okay, should be good for (hopefully final?) review. I moved the tracing init stuff to CJS modules. It seems to be the least headache to use right now in the sample apps I'm using as I document stuff. Once ESM is properly supported in the SDK we'll probably want to figure out a way to show both options and instructions for running.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 14]
```diff
@@ -3,135 +3,211 @@ title: Instrumentation
...
+});
+
+provider.register();
+
+module.exports.tracer = opentelemetry.trace.getTracer(
+  'name-of-tracer-goes-here'
+);
+```
+
+Next, ensure that `tracing.js` is required in your node invocation. For example:
```
Comment (austinlparker) [inline content/en/docs/instrumentation/js/instrumentation.md:56]: Would it be worth a callout here to note that `tracing` should be imported before any libraries targeted by auto-instrumentation libraries?

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 15]
```diff
@@ -188,37 +258,36 @@ const { SemanticAttributes } = require('@opentelemetry/semantic-conventions');
...
 ```js
 span.addEvent('Doing something');
 
 const result = doWork();
-
-span.addEvent('Did something');
 ```
 
-You can also add an object with more data to go along with the message:
+You can also create Span Events with additiona
```
Comment (austinlparker) [inline content/en/docs/instrumentation/js/instrumentation.md:289]:
```suggestion
You can also create Span Events with additional
```

**Review by austinlparker — APPROVED** [review]
lgtm outside of some nits

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 9 above)
Comment (gareththackeray) [reply content/en/docs/instrumentation/js/instrumentation.md:350]: Could you elaborate on why that is?  Wouldn't the application dev know whether the span completed or not?

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 16]
```diff
@@ -3,135 +3,211 @@ title: Instrumentation
...
+To initialize tracing with the Node.js SDK, first ensure you have the SDK
+package and OpenTelemetry API installed:
+
+```shell
+npm install \
+  @opentelemetry/api \
+  @opentelemetry/sdk-trace-node
+```
+
+Next, create a separate `tracing.js` file that has all the SDK initialization
```
Comment (svrnm) [inline content/en/docs/instrumentation/js/instrumentation.md:30]: I am just wondering why the code in `tracing.js` is separated out from the application itself, what's the goal here? Make it easier to swap out the SDK with a vendor specific one? 

If so, the sample app still needs to load the API to work without the `-r tracing.js` I assume? 

This will also be affected from the issue https://github.com/open-telemetry/opentelemetry-js/issues/2912. 

The sample app is also not lined out in the in the document anymore, which makes it hard for the enduser to follow those steps by simply doing a "copy&paste".

@dyladan you know this probably best, any thoughts?

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 17]
```diff
@@ -3,135 +3,211 @@ title: Instrumentation
...
-  console.log('work...');
-  // simulate some random work.
-  for (let i = 0; i <= Math.floor(Math.random() * 40000000); i += 1) {}
-}
+```
+node --require './tracing.js' <app-file.js>
 ```
 
-## Initializing a Tracer
+### Browser
```
Comment (svrnm) [inline content/en/docs/instrumentation/js/instrumentation.md:62]: This looks similar to https://opentelemetry.io/docs/instrumentation/js/getting-started/browser/, is the goal to replace that one and have browser/node.js in a single file?

**Review by svrnm — CHANGES_REQUESTED** [review]
Added a few comments, my main concerns are the following:

* by removing the example application it's much harder to follow the flow
* see my comment on `tracing.js`, it's not clear why the code is separated out like for auto instrumentation (where a "don't touch my code" experience is nice). If the goal is to separate SDK & API this should be called out to not confuse the user.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 18]
```diff
@@ -3,135 +3,211 @@ title: Instrumentation
...
+Next, create a separate `tracing.js` file that has all the SDK initialization
+code in it:
+
+```js
+const opentelemetry = require("@opentelemetry/api");
+const { NodeTracerProvider } = require("@opentelemetry/sdk-trace-node");
+const { registerInstrumentations } = require("@opentelemetry/instrumentation");
+const { ConsoleSpanExporter, BatchSpanProcessor } = require("@opentelemetry/sdk-trace-base");
+
+const provider = new NodeTracerProvider();
```
Comment (dyladan) [inline content/en/docs/instrumentation/js/instrumentation.md:39]: I would include a resource that has at least a service name since that is a required attribute

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 14 above)
Comment (dyladan) [reply content/en/docs/instrumentation/js/instrumentation.md:56]: Yes I think it is worth it. That is why we use the --register/-r flag

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 19]
```diff
@@ -3,135 +3,211 @@ title: Instrumentation
...
+provider.addSpanProcessor(processor);
+
+// Optionally register automatic instrumentation libraries
+registerInstrumentations({
+  instrumentations: [],
+});
+
+provider.register();
+
+module.exports.tracer = opentelemetry.trace.getTracer(
```
Comment (dyladan) [inline content/en/docs/instrumentation/js/instrumentation.md:51]: Instead of exporting the tracer I think it would be better to encourage the user to use api.trace.getTracer. If it is exported from here then the main app.js will need to require this file and we may run into odd application load ordering problems if for example the user requires other dependencies first

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 16 above)
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:30]: I've seen this as a pretty common thing I've seen, to isolate all initialization stuff (and instrumentation registration) into a separate file and then just use the API in the other files in an app where manual instrumentation is done. I'm fine with changing this; whatever comports most with how you'd initialize opentelemetry in a real-world app.

As for a sample, I could also add it back in, but unlike a getting-started flow this doc tends to be more of a "I know I want to do a thing, let me quickly find that subtopic and let me copy-paste some code into my own app and then tweak it so it compiles/runs as-needed" sort of thing. I don't think having a single sample that you run through from top to bottom is as important here.

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 20]
```diff
@@ -292,3 +363,83 @@ try {
...
+// This is what we'll access in all instrumentation code
+export const tracer = opentelemetry.trace.getTracer(
+  'example-basic-tracer-node'
+);
+```
+
+Like the other examples in this document, this exports a tracer you can use
+throughout the app.
+
+## Creating nested spans
```
[the remark this answers is not in the document]
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:401]: It is not, I'll clarify in the title. Creating nested spans is harder in the non-Node/non-browser SDK case, because you need to manually set the current span in context before creating a nested span when using only `sdk-trace-base`. And we should definitely not make that the default recommendation, since it's the non-majority case and more code.

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 17 above)
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:62]: The goal is to show both of the major ways to initialize tracing. Since each uses a different SDK, I think it's better to have both sections here.

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 16 above)
Comment (Flarna) [reply content/en/docs/instrumentation/js/instrumentation.md:33]: One very common pitfall regarding instrumentations is that it's needed to first create/register the OTel instrumentation and then require the corresponding module.

By using a dedicated file which is preloaded (or required at the beginning of of the main script) this works more or less automatic.
If you put the OTel init code into your main script one ends up with a Otel section (require + code) and then again 
some require lines and code. Most people don't like require lines after code sections so they tend to resort it - and break their OTel setup.

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 17 above)
Comment (svrnm) [reply content/en/docs/instrumentation/js/instrumentation.md:74]: OK. Should we then get rid of the browser getting started?

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (svrnm) [reply content/en/docs/instrumentation/js/instrumentation.md]: Ah, makes sense, then the issue is the heading level, it probably should be `###`

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 17 above)
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md:74]: I don't think so, no. The goal here is not to be DRY. I think it's fine to have some duplicated code in different contexts.

**Comment by chalin** [comment]
@svrnm - this good to go?

**Comment by svrnm** [comment]
> @svrnm - this good to go?

There are still 2 things open for me:

- minor: `Creating nested spans with sdk-trace-base` needs to be a level 3 heading
- major: I still think that having a sample app people can use to try things out would be [valuable](https://github.com/open-telemetry/opentelemetry.io/pull/1445#discussion_r906001719), I am not going to die on that hill if I am outvoted, but I'd like to have that clarified.

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (cartermp) [reply content/en/docs/instrumentation/js/instrumentation.md]: Thanks, missed this!

**Comment by dyladan** [comment]
I appreciate the value of a sample app, but I think this PR is already pretty large and is already such an improvement over the existing document that it should be accepted and a sample can be taken as a follow up task.

**Comment by chalin** [comment]
> I appreciate the value of a sample app, but I think this PR is already pretty large and is already such an improvement over the existing document that it should be accepted and a sample can be taken as a follow up task.

+1

@svrnm - your first point has been addressed, and I agree iwth @dyladan that further changes should be addresses separately. Might you be willing to approve this PR as is?

File: content/en/docs/instrumentation/js/instrumentation.md (hunk 16 above)
Comment (vmarchaud) [reply content/en/docs/instrumentation/js/instrumentation.md:33]: > Maybe it's worthwhile do have a section explaining exactly that, i.e. using exactly your words why you might like to fork out otel setup into it's own file and then either require it via -r tracing.js or a `require("tracing.js") statement?

I've made a similar comment on the other PR but i understand that we want to avoid to overwhelm readers with too much "gotcha" to know. However it might be worth to add a Q&A section later on to document these

File: content/en/docs/instrumentation/js/instrumentation.md [hunk 21]
```diff
@@ -3,135 +3,255 @@ title: Instrumentation
...
+
+const provider = new NodeTracerProvider({
+    resource: resource,
+});
+const exporter = new ConsoleSpanExporter();
+const processor = new BatchSpanProcessor(exporter);
+provider.addSpanProcessor(processor);
+
+// Optionally register automatic instrumentation libraries
+registerInstrumentations({
```
Comment (vmarchaud) [inline content/en/docs/instrumentation/js/instrumentation.md:60]: i would move the `registerInstrumentations` before the initialization of the exporter (L55) since 3rd party exporter might require dependency before the instrumentation can patch it (specially http)

**Review by vmarchaud — COMMENTED** [review]
overall lgtm, nit pick that would better if people modify the init script later on though

**Review by svrnm — APPROVED** [review]

**Comment by cartermp** [comment]
I'd also like to think about what a good sample looks like such that each subsection could be meaningfully applied in a way that doesn't feel arbitrary/strange. It might also be good to settle if we want to do something like that for all the manual instrumentation pages. Currently, none of them work off of a common sample/scenario.

**Comment by svrnm** [comment]
I closed my outstanding conversations as well, so we can move that "sample app" discussion to an issue for now.

**Review by chalin — APPROVED** [review]
Thanks @cartermp & all for your efforts!
